src/Mock.spec.ts

```diff
@@ -282,6 +302,51 @@ describe('jest-mock-extended', () => {
         });
     });
 
+    describe('Deep mock support for class variables which are functions but also have nested properties and functions', () => {
+        test('can deep mock members', () => {
+            const mockObj = mockDeep<Test6>();
+            const input = new Test1(1);
+            mockObj.funcValueProp.nonDeepProp.calledWith(input).mockReturnValue(4);
+
+            expect(mockObj.funcValueProp.nonDeepProp(input)).toBe(4);
+        });
+
+        test('three or more level deep mock', () => {
+            const mockObj = mockDeep<Test6>();
+            mockObj.funcValueProp.deepProp.deeperProp.getNumber.calledWith(1).mockReturnValue(4);
+
+            expect(mockObj.funcValueProp.deepProp.deeperProp.getNumber(1)).toBe(4);
+        });
+
+        test('maintains API for deep mocks', () => {
+            const mockObj = mockDeep<Test6>();
+            mockObj.funcValueProp.deepProp.getNumber(100);
+
+            expect(mockObj.funcValueProp.deepProp.getNumber.mock.calls[0][0]).toBe(100);
+        });
+
+        test('deep expectation work as expected', () => {
+            const mockObj = mockDeep<Test6>();
+            mockObj.funcValueProp.deepProp.getNumber(2);
+
+            expect(mockObj.funcValueProp.deepProp.getNumber).toHaveBeenCalledTimes(1);
+        });
+
+        test('can mock base function which have properties', () => {
```

Comment:
This test and the test below this are the ones that actually test the new behaviour